packages/core/src/components/Modal/private/InnerContent.tsx

```diff
@@ -11,40 +12,152 @@ export type Props = {
+  medium?: boolean;
+  /** Modal content height becomes scrollable. */
+  scrollable?: boolean;
+  /** Dialog header subtitle. */
+  subtitle?: React.ReactNode;
   /** Dialog header title. */
   title?: React.ReactNode;
   /** Callback for when the Dialog should be closed.  */
   onClose: (event: React.MouseEvent<any> | React.KeyboardEvent) => void;
 };
 
 /** A Dialog component with a backdrop and a standardized layout. */
-export default function ModalInnerContent({ children, footer, large, onClose, title }: Props) {
+function ModalInnerContent({
+  children,
+  cx,
+  footer,
+  large,
+  medium,
+  onClose,
+  scrollable,
+  styles,
+  subtitle,
+  title,
+}: Props & WithStylesProps) {
   const theme = useTheme();
+  const withHeader = Boolean(title || subtitle);
+  const withFooter = Boolean(footer);
 
   return (
-    <Spacing bottom={6} horizontal={4} top={4}>
-      <Spacing bottom={title ? 3 : 0} tag="header">
-        <Spacing bottom={4}>
-          <IconButton onClick={onClose}>
-            <IconClose
-              accessibilityLabel={T.phrase('Close', {}, 'Close a modal popup')}
-              size={3 * theme.unit}
-            />
-          </IconButton>
-        </Spacing>
-
-        {title && <Title level={large ? 1 : 3}>{title}</Title>}
-      </Spacing>
-
-      {children}
+    <div className={cx(styles.wrapper)}>
+      {withHeader && (
+        <header className={cx(styles.header, scrollable && styles.header_scrollable)}>
+          {title && <Title level={3}>{title}</Title>}
+          {subtitle && <Text muted>{subtitle}</Text>}
+        </header>
+      )}
+
+      <div className={cx(styles.close, !withHeader && styles.close_float)}>
+        <IconButton onClick={onClose}>
+          <IconClose
+            accessibilityLabel={T.phrase('Close', {}, 'Close a modal popup')}
+            color={theme.color.muted}
+            size={theme.unit * 3}
+          />
+        </IconButton>
+      </div>
+
+      <div
+        className={cx(
+          styles.body,
+          !withHeader && styles.body_paddingTop,
+          !withFooter && styles.body_paddingBottom,
+          scrollable && styles.body_scrollable,
+          scrollable && (medium || large) && styles.body_scrollableLarge,
+        )}
+      >
+        {children}
+      </div>
 
       {footer && (
-        <Spacing tag="footer" top={4}>
+        <footer className={cx(styles.footer, scrollable && styles.footer_scrollable)}>
           {footer}
-        </Spacing>
+        </footer>
       )}
-    </Spacing>
+    </div>
   );
 }
+
+export default withStyles(({ color, ui, unit }) => ({
+  wrapper: {
+    position: 'relative',
+  },
+
+  header: {
+    padding: unit * 3,
+  },
+
+  header_scrollable: {
+    position: 'relative',
+
+    ':after': {
+      content: '" "',
+      position: 'absolute',
+      top: '100%',
+      left: 0,
+      right: 0,
+      height: unit / 2,
+      background: 'linear-gradient(rgba(0, 0, 0, .1), rgba(0, 0, 0, 0))',
```

Comment:
to match design prototype and spec

Reply:
Not sure how well this works in dark mode.

Reply:
hrmm suggestions? toRGB helper from whatever color.core.bg is?

Reply:
Yeah most likely.

Reply:
that's the closest i could get to black. so had to up the opacity to get close to the original visuals of black 10%